awx/main/models/jobs.py

```diff
@@ -314,10 +331,16 @@ def get_effective_slice_ct(self, kwargs):
         actual_inventory = self.inventory
         if self.ask_inventory_on_launch and 'inventory' in kwargs:
             actual_inventory = kwargs['inventory']
+        actual_slice_count = self.job_slice_count
+        if self.ask_job_slice_count_on_launch and 'job_slice_count' in kwargs:
+            actual_slice_count = kwargs['job_slice_count']
+            # Set the eager fields if its there as well
+            if '_eager_fields' in kwargs and 'job_slice_count' in kwargs['_eager_fields']:
+                kwargs['_eager_fields']['job_slice_count'] = actual_slice_count
```

Comment:
As a coding practice... obviously I would not like a `get_*` method to _set_ something like this.

But it's much more troublesome that there was some reason that caused this to need to get set here in the first place. This would seem to indicate that `job_slice_count` doesn't get copied on launch to the job. This should happen in `create_unified_job` irrelevant of the prompts.

```
In [8]: set(JobTemplate.get_ask_mapping().keys()) - set(WorkflowJobTemplate._get_unified_job_field_names())
Out[8]: 
{'credentials',
 'diff_mode',
 'execution_environment',
 'forks',
 'instance_groups',
 'job_slice_count',
 'job_type',
 'timeout',
 'verbosity'}
```

I expect some of these probably need to be added to the workflow `_get_unified_job_field_names`, because they are valid workflow fields.